getting_started/step_by_step/resources.rst

```diff
@@ -158,12 +158,259 @@ Freeing resources
+            obj1.free()                       # obj2's Dictionary still exists
+            print(obj2.dict.greeting)         # prints 'hello'
+
+Godot makes it easy to create custom Resources in the Inspector.
+
+1. Create a plain Resource object in the Inspector. This can even be a type that derives Resource, so long as your script is extending that type.
+2. Set the ``script`` property in the Inspector to be your script.
+
+The Inspector will now display your Resource script's custom properties. If one edits
+those values and saves the resource, the Inspector serializes the custom properties
+too! To save a resource from the Inspector, click the Inspector's tools menu (top right),
+and select "Save" or "Save As...".
+
+If the script's language supports `script classes <https://godot.readthedocs.io/en/latest/getting_started/step_by_step/scripting_continued.html#register-scripts-as-classes>`__, 
+then it streamlines the process. Defining a name for your script alone will add it to
+the Inspector's creation dialog. This will auto-add your script to the Resource
+object you create.
+
+Let's see some examples.
+
+.. tabs::
+  .. code-tab:: gdscript GDScript
+
+    # bot_stats.gd
+    extends Resource
+    export(int) var health
+    export(Resource) var sub_resource
+    export(Array, String) var str_array
+
+    func _init(p_health = 0, p_sub_resource = null, p_str_array = []):
+        health = p_health
+        sub_resource = p_sub_resource
+        str_array = p_str_array
+
+    # bot.gd
+    extends KinematicBody
+    export(Resource) var stats
+
+    const BotStats = preload("res://bot_stats.gd")
+
+    func _init():
+        if not stats:
+            stats = BotStats.new(10)
+    
+    func _ready():
+        print(stats.health) # prints '10'
+  .. code-tab:: csharp
+
+        // BotStats.cs
+        using System;
+        using Godot;
+
+        namespace ExampleProject {
+            public class BotStats : Resource
+            {
+                [Export]
+                public int Health { get; set; }
+
+                [Export]
+                public Resource SubResource { get; set; }
+
+                [Export]
+                public String[] Strings { get; set; }
+
+                public override void _Init(int health = 0, Resource subResource = null, String[] strings = [])
+                {
+                    Health = health;
+                    SubResource = subResource;
+                    Strings = strings;
+                }
+            }
+        }
+
+        // Bot.cs
+        using System;
+        using Godot;
+
+        namespace ExampleProject {
+            public class Bot : KinematicBody
+            {
+                [Export]
+                public BotStats Stats;
+
+                public override void _Init()
+                {
+                    if (!Stats)
+                    {
+                        Stats = (ResourceLoader::load("BotStats.cs") as BotStats).new(10);
```

Comment:
If we are using the constructor as suggested above we can use:

    stats = new BotStats(10);